client/model/cluster.go

```diff
@@ -107,14 +118,15 @@ type AwsAttributes struct {
 
 // DbfsStorageInfo contains the destination string for DBFS
 type DbfsStorageInfo struct {
-	Destination string `json:"destination,omitempty"`
+	Destination string `json:"destination"`
 }
 
 // S3StorageInfo contains the struct for when storing files in S3
 type S3StorageInfo struct {
-	Destination      string `json:"destination,omitempty"`
-	Region           string `json:"region,omitempty"`
-	Endpoint         string `json:"endpoint,omitempty"`
+	// TODO: add instance profile validation check + prefix validation
+	Destination      string `json:"destination"`
+	Region           string `json:"region,omitempty" tf:"group:location"`
+	Endpoint         string `json:"endpoint,omitempty" tf:"group:location"`
```

Comment:
what does the group keyword signify?

Reply:
If it is meant to be a block, we should think about renaming it to block; https://www.terraform.io/docs/configuration/terraform.html#terraform-block-syntax

Reply:
group means "group of mutually exclusive parameters". something to simplify validation